src/main/java/io/jsonwebtoken/impl/DefaultJwtParser.java

```diff
@@ -58,6 +62,54 @@
 
     private SigningKeyResolver signingKeyResolver;
 
+    Claims expectedClaims = new DefaultClaims();
+
```

Comment:
Is this cheating? Date gets handled differently internally, so I did the null check and then used the specific setter.

All the other claims are String based, so I just call `expect`. This makes it so I don't have to do a null check for each type.